hoot-core/src/main/cpp/hoot/core/conflate/NodeMatcher.cpp

```diff
@@ -195,13 +211,25 @@ double NodeMatcher::scorePair(long nid1, long nid2)
   }
   else
   {
-    vector<bool> exclude(theta1.size(), false);
-    thetaScore = _calculateAngleScore(theta1, theta2, exclude, 0);
+    if (theta2.size() < theta1.size())
+    {
+      vector<bool> exclude(theta2.size(), false);
+      thetaScore = _calculateAngleScore(theta2, theta1, exclude, 0, nid1 == -2831);
```

Comment:
I'm probably jumping at shadows, but seeing nid1 == -2831 makes my spider-programmer-sense tingle. How special is -2831?